compiler/dyno/tools/chpldoc/chpldoc.cpp

```diff
@@ -797,49 +825,73 @@ struct RstResultBuilder {
   Context* context_;
   std::stringstream os_;
   std::vector<RstResult*> children_;
-
+  const std::string commentStyle = commentStyle_;
   static const int commentIndent = 3;
 
-  void showComment(const Comment* comment, bool indent=true) {
-    if (!comment) {
+  bool showComment(const Comment* comment, bool indent=true) {
+    if (!comment || comment->str().substr(0, 2) == "//") {
       os_ << '\n';
-      return;
+      return false;
     }
 
     int indentChars = indent ? commentIndent : 0;
-    auto lines = prettifyComment(comment->str());
-    os_ << '\n';
+    auto lines = prettifyComment(comment->str(), commentStyle);
+    if (!lines.empty())
+      os_ << '\n';
     for (const auto& line : lines) {
       if (line.empty()) {
         os_ << '\n';
       } else {
         indentStream(os_, indentChars) << line << '\n';
       }
     }
-    os_ << '\n';
+    os_ << "\n";
+    return true;
   }
-  void showComment(const AstNode* node, bool indent=true) {
-    showComment(previousComment(context_, node->id()), indent);
+  bool showComment(const AstNode* node, bool indent=true) {
+    return showComment(previousComment(context_, node->id()), indent);
   }
 
   template<typename T>
-  void show(const std::string& kind, const T* node, bool indentComment=true) {
-    if (isNoDoc(node)) return;
+  bool show(const std::string& kind, const T* node, bool indentComment=true) {
+    if (isNoDoc(node)) return false;
 
     os_ << ".. " << kind << ":: ";
     RstSignatureVisitor ppv{os_};
     node->traverse(ppv);
     os_ << "\n";
 
-    showComment(node, indentComment);
+    bool commentShown = showComment(node, indentComment);
+
+    // TODO: Fix all this because why are we checking for specific node type here?
+    if (commentShown && node->isEnum()) {
+      os_ << "\n";
+    }
 
     if (auto attrs = node->attributes()) {
       if (attrs->isDeprecated()) {
-        indentStream(os_, 1 * indentPerDepth) << ".. warning::\n";
-        indentStream(os_, 2 * indentPerDepth) << attrs->deprecationMessage().c_str();
-        os_ << "\n\n";
+        auto comment = previousComment(context_, node->id());
+        if (comment && !comment->str().empty() && comment->str().substr(0, 2) == "/*" &&
```

Comment:
Up to you, but this might be nice to capture in some kind of "isDocComment" helper function.

Reply:
Thanks, I'll make a note to fix it in a later PR.